src/components/Layout/utils.ts

```diff
@@ -0,0 +1,121 @@
+import cn from '@ably/ui/core/utils/cn';
+import { AccordionProps } from '@ably/ui/core/Accordion';
+import { ProductData, ProductKey } from 'src/data/types';
+import { NavProductContent, NavProductPage, NavProductPages } from 'src/data/nav/types';
+
+type ActivePage = { tree: number[]; page: NavProductPage };
+
+/**
+ * Determines the active page based on the provided target link.
+ *
+ * This function iterates through the product data to find a page that matches the target link.
+ * It checks both direct links and nested pages within groups. If a match is found, it returns
+ * the index tree of the page in the navigation data and the page details.
+ *
+ * @param data - The product data containing navigation information.
+ * @param targetLink - The link to be matched against the product data.
+ * @returns The active page details including its tree and page information, or null if no match is found.
+ */
+export const determineActivePage = (data: ProductData, targetLink: string): ActivePage | null => {
+  const strippedTargetLink = formatNavLink(targetLink);
+
+  /**
+   * Recursively determines the presence of a page in a list of pages.
+   *
+   * This function iterates through the list of pages and checks if the target link is present.
+   * If a match is found, it returns the index of the link in its pages array.
+   * If not, it attempts to search for the link in nested pages.
+   */
+  const determinePagePresence = (pages: NavProductPages[], path: number[]): number[] | null => {
+    for (let i = 0; i < pages.length; i++) {
+      const page = pages[i];
+
+      // If the page is a link and the link matches the target link, return the path
+      if ('link' in page && formatNavLink(page.link) === strippedTargetLink) {
+        return [...path, i];
+      }
+
+      // If the page is a group of pages, recursively check each page
+      if ('pages' in page && page.pages) {
+        const result = determinePagePresence(page.pages, [...path, i]);
+        if (result) {
+          return result;
+        }
+      }
+    }
+    return null;
+  };
+
+  for (const key of Object.keys(data) as ProductKey[]) {
+    /* 
+      If the page matches a product index link, return details for the product
+    */
+    if (data[key].nav.link === strippedTargetLink) {
+      const { name, link } = data[key].nav;
```

Comment:
the link might be undefined? shall we add a default string?

`const { name, link = '' } = data[key].nav;`